Modules/Options/GeneralTab/QuestieOptionsGeneral.lua

```diff
@@ -663,19 +683,35 @@ function QuestieOptions.tabs.general:Initialize()
             },
             objectiveCompleteSoundButton = {
                 type = "execute",
-                order = 2.14,
+                order = 2.15,
                 name = "",
                 width = 0.5,
                 image = function ()
                     return "Interface\\OptionsFrame\\VoiceChat-Play", 15, 15
                 end,
                 func = function ()
-                    PlaySoundFile(Sounds.QUEST_OBJECTIVE_COMPLETE_SOUND_FILE)
+                    local soundFile = Questie.db.char.objectiveCompleteSoundChoice ~= nil and Questie.db.char.objectiveCompleteSoundChoice or Sounds.QUEST_COMPLETE_SOUND_FILE
+                    PlaySoundFile(soundFile, "Master")
                 end
             },
+            objectiveCompleteSoundChoice = {
+                type = "select",
+                order = 2.16,
+                values = _GetObjectiveSoundChoices(),
+                sorting = _GetObjectiveSoundChoicesSort(),
+                style = 'dropdown',
+                name = function() return l10n('Objective Complete Sound Selection') end,
+                desc = function() return l10n('The sound you hear when an objective is completed'); end,
+                get = function() return  Questie.db.char.objectiveCompleteSoundChoiceName; end,
+                disabled = function() return (not Questie.db.char.soundOnObjectiveComplete); end,
+                set = function(input, value)
+                    Questie.db.char.objectiveCompleteSoundChoiceName = value
+                    Questie.db.char.objectiveCompleteSoundChoice = Questie.sounds[value]
```

Comment:
Instead of storing two values we can only store `Questie.db.char.objectiveCompleteSoundChoiceName` and call `Questie.sounds[Questie.db.char.objectiveCompleteSoundChoiceName]` when ever we need a sound. This could also be wrapped in a new function in the Sounds.lua 🤔

Reply:
new function?

Reply:
I was thinking about a call to a new:

```lua
function Sounds.GetSelectedSoundFile()
    return Questie.sounds[Questie.db.char.objectiveCompleteSoundChoiceName]
end
```
 
That way the other modules don't have to know about the actual structure of `Questie.db.char.objectiveCompleteSoundChoiceName`. In addition that might encourage to replace `Questie.sounds` with a local table inside the Sounds module.

Reply:
fixed.. take a look now?